sdks/python/apache_beam/testing/synthetic_pipeline.py

```diff
@@ -238,13 +241,25 @@ def get_range_tracker(self, start_position, stop_position):
       tracker = range_trackers.UnsplittableRangeTracker(tracker)
     return tracker
 
+  def _gen_kv_pair(self, index):
+    r = np.random.RandomState(index)
+    rand = r.random_sample()
+
+    # Determines whether to generate hot key or not.
+    if rand < self._hot_key_fraction:
+      # Generate hot key.
+      # An integer is randomly selected from the range [0, numHotKeys-1]
+      # with equal probability.
+      r_hot = np.random.RandomState(self._num_hot_keys)
```

Comment:
Doesn't this generate the key from the same seed always? So it's always a single key? Maybe using something like `index % num_hot_keys`? Or something like that..

Reply:
You are right. Or maybe just `index`?

Reply:
If we use `index`, then we will have many different seeds, so we'll jave many different 'hot keys' instead of restricting to a total of `num_hot_keys` - I think?

Reply:
That totally makes sense, thanks a lot! And it also answers my second comment here, I guess :) I'll change it this way.